wallet-new/src/Cardano/Wallet/API/V1/Types.hs

```diff
@@ -316,6 +321,19 @@ deriveJSON Serokell.defaultOptions ''Address
 instance Arbitrary Address where
   arbitrary = Address . fromString <$> elements ["DEADBeef", "123456"]
 
+-- | Whether an address is valid or not.
+newtype AddressValidity = AddressValidity { getValid :: Bool }
+  deriving (Eq, Show, Generic)
+
+deriveJSON Serokell.defaultOptions ''AddressValidity
```

Comment:
I'm afraid the stock `deriveJSON` instance won't cut it, as users would receive this slightly-warty JSON:

```haskell
{ 
  "getValid": true
}
```

Whereas we could have something like:

```haskell
{ 
  "valid" : true
}
```

Which looks better (at least to my eyes, which I appreciate might not be the gold standard 😁  )

Reply:
Are you sure this happens with the Serokell defaultOptions? I looked here: https://github.com/serokell/serokell-util/blob/9f9715763134e41c9ba92a1359be92f97f93eea8/src/Serokell/Aeson/Options.hs. It looks as if the generated JSON should end up being `{ "valid" : true }` unless I am confused?

Reply:
I think you might be right -- although it doesn't play very well with the `getXX` convention, sigh.